// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.stylelens.detect";
option java_outer_classname = "ObjectDetectProto";
option objc_class_prefix = "STYL";

package objectdetect;

service Detect{
  rpc GetObjects(DetectRequest) returns (stream DetectReply) {}
}

message Location {
  float left = 1;
  float right = 2;
  float top = 3;
  float bottom = 4;
}

message DetectRequest {
  bytes file_data = 1;
}

message DetectReply {
    Location location = 1;
    string class_name = 2;
    string class_code = 3;
    bytes feature = 4;
}
